projects/packages/changelogger/tests/php/tests/src/Plugins/WordpressVersioningTest.php

```diff
@@ -289,6 +292,11 @@ public function provideCompareVersions() {
 			array( '1.1.1-beta9', '==', '1.1.1-beta9' ),
 			array( '1.1.1-alpha2', '<', '1.1.1-alpha10' ),
 			array( '1.1.1+beta.9.1', '==', '1.1.1+beta.9' ),
+			array( '10.2-2109.1', '>', '10.1' ),
+			array( '10.2-2109.1', '<', '10.2' ),
+			array( '10.2-2109.1', '>', '10.2-alpha' ),
+			array( '10.2-2109.1', '<', '10.2-beta' ),
+			array( '10.2-2109.2', '<', '10.2-2109.10' ),
 		);
```

Comment:
This will fail a test like
```suggestion
			array( '10.2-2112.2', '<', '10.2-2201.1' ),
		);
```

Yeah, in practice we'd probably have made it '10.3-2201.1' instead, but we should still do the comparison correctly in case someday we skip a month or something.

The fix will probably have to involve turning the "count" from `parsePrerelease()` into an array of "counts". Or we could change the new regex (in both places) to be something like `\d\d(?:0[1-9]|1[0-2])\.\d{1,4}` (limiting the second part to 4 digits max) so we can capture both numbers and make the "count" be like `$m[1] * 10000 + $m[2]`.

Reply:
> The fix will probably have to involve turning the "count" from parsePrerelease() into an array of "counts".

For funsies I've given this approach a shot in: https://github.com/Automattic/jetpack/pull/21140/commits/0c4c757f9ed914228452fe8170ca9ee6cc8cf50b

Not sure if doing `if ( 2 === $aindex && 2 === $bindex )` is weird inside of `compareVersions()`, or if there is a better way to handle this new pattern.